ambry-store/src/main/java/com.github.ambry.store/DiskManager.java

```diff
@@ -282,6 +296,22 @@ boolean shutdownBlobStore(PartitionId id) {
     }
   }
 
+  /**
+   * Set the BlobStore stopped state with given {@link PartitionId} {@code id}.
+   * @param id the {@link PartitionId} of the {@link BlobStore} whose stopped state should be set.
+   * @param isStopped whether to mark BlobStore as stopped ({@code true}) or started.
+   * @return {@code true} if set stopped state of store was successful. {@code false} if not.
+   */
+  boolean setBlobStoreStoppedState(PartitionId id, boolean isStopped) {
+    BlobStore store = stores.get(id);
+    if (store == null) {
+      // no need to check if the store is started because this method could be called after store is successfully shutdown.
+      logger.error("store is not found on this disk when trying to update stoppedReplicas list");
+      return false;
+    }
+    return store.updateStoppedReplicasList(isStopped);
```

Comment:
to follow up on @vgkholla's earlier comment, why relay this via the `BlobStore`? Can't this directly call into the `ReplicaStatusDelegate`?

Reply:
The original purpose of introducing "STOPPED" list is to mark specific BlobStore as `stopped` when we manually "stop" the BlobStore. So the update operation is performed at store level. Note that the `markStopped` operation is initiated in `AmbryRequests` and we don't want to expose blobstore to `AmbryRequests`, that's why we chose to relay operation through StorageManager and DiskManager. Also, we currently don't have `ReplicaStatusDelegate` in both `StorageManager` and `DiskManager`, which disallows us to directly call methods in `ReplicaStatusDelegate`.  (Maybe we can add `ReplicaStatusDelegate` as a field in StorageManager/DiskManager later)